explorer_frontend/src/features/account-connector/components/EndpointScreen.tsx

```diff
@@ -0,0 +1,215 @@
+  const [inputValue, setInputValue] = useState("");
+  const [error, setError] = useState("");
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  // Get initialization states
+  const [
+    smartAccount,
+    balance,
+    balanceToken,
+    initializingSmartAccountState,
+    initializingSmartAccountError,
+  ] = useUnit([
+    $smartAccount,
+    $balance,
+    $balanceToken,
+    $initializingSmartAccountState,
+    $initializingSmartAccountError,
+  ]);
+
+  const [isPendingSmartAccountCreation] = useUnit([createSmartAccountFx.pending]);
+
+  const animationContainerRef = useRef<HTMLDivElement | null>(null);
+
+  // Initialize Lottie animation on mount
+  useEffect(() => {
+    if (animationContainerRef.current) {
+      const animationInstance = lottie.loadAnimation({
+        container: animationContainerRef.current as Element,
+        renderer: "svg",
+        loop: true,
+        autoplay: true,
+        animationData,
+      });
+
+      return () => animationInstance.destroy();
+    }
+  }, [isPendingSmartAccountCreation]);
+
+  // Handles connect button press
+  const handleConnect = () => {
+    const validation: ValidationResult = validateRpcEndpoint(inputValue);
+    if (!validation.isValid) {
+      setError(validation.error);
+      return;
+    }
+
+    setError("");
+    setIsDisabled(true);
+
+    setEndpoint(inputValue);
+    initilizeSmartAccount();
+  };
+
+  // Opens a new browser tab to fetch the endpoint URL
+  const handleGetEndpoint = () => {
+    const endpointUrl = import.meta.env.VITE_GET_ENDPOINT_URL;
+    if (endpointUrl) {
+      window.open(endpointUrl, "_blank");
+    } else {
+      console.error("VITE_GET_ENDPOINT_URL is not set.");
+    }
+  };
+
+  // Clears error message when user types
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setError("");
+    setInputValue(e.target.value);
+  };
+
+  // **Logic to switch to MainScreen**
+  useEffect(() => {
+    if (smartAccount !== null && balance !== null && balanceToken !== null) {
+      setActiveComponent(ActiveComponent.Main);
+    }
+  }, [smartAccount, balance, balanceToken]);
+
+  // **Reset UI if error occurs**
+  useEffect(() => {
+    if (initializingSmartAccountError) {
+      setIsDisabled(false); // Re-enable everything
+      setError(initializingSmartAccountError);
+    }
+  }, [initializingSmartAccountError]);
+
+  return (
+    <div
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        justifyContent: "space-between",
+        textAlign: "center",
+        minHeight: "400px",
+      }}
+    >
+      {/* While creating account, show animation & state instead of logo and heading */}
+      {!isPendingSmartAccountCreation ? (
+        <div style={{ width: "100%" }}>
+          <img src={asteriskIcon} alt="Asterisk Icon" width={124} height={124} />
+          <HeadingLarge>Enter the RPC URL to connect the wallet</HeadingLarge>
```

Comment:
why do we ask on pending to enter rpc url?

Reply:
!pending